internal/oci/oci.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oci
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/sigstore/cosign/cmd/cosign/cli/fulcio"
+	"github.com/sigstore/cosign/cmd/cosign/cli/rekor"
+	ociremote "github.com/sigstore/cosign/pkg/oci/remote"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	coptions "github.com/sigstore/cosign/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/sigstore/cosign/pkg/oci"
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
+	"github.com/sigstore/sigstore/pkg/signature"
+)
+
+// options is a struct that holds options for verifier.
+type options struct {
+	PublicKey []byte
+	Keychain  authn.Keychain
+}
+
+// Options is a function that configures the options applied to a Verifier.
+type Options func(opts *options)
+
+// WithPublicKey sets the public key.
+func WithPublicKey(publicKey []byte) Options {
+	return func(opts *options) {
+		opts.PublicKey = publicKey
+	}
+}
+
+func WithAuthnKeychain(keychain authn.Keychain) Options {
+	return func(opts *options) {
+		opts.Keychain = keychain
+	}
+}
+
+// Verifier is a struct which is responsible for executing verification logic.
+type Verifier struct {
+	opts *cosign.CheckOpts
+}
+
+// New initializes a new Verifier.
+func New(ctx context.Context, opts ...Options) (*Verifier, error) {
```

Comment:
Maybe this should be called `NewVerifier` because the package name, `oci`, doesn't reflect what `New()` would provide.